src/modules/dashboard/sagas/motions/finalizeMotion.ts

```diff
@@ -26,39 +24,22 @@ function* finalizeMotion({
   try {
     const colonyManager = TEMP_getContext(ContextModule.ColonyManager);
     const { provider } = colonyManager;
-    const colonyClient = yield colonyManager.getClient(
-      ClientType.ColonyClient,
-      colonyAddress,
-    );
     // @NOTE This line exceeds the max-len but there's no prettier solution
     // eslint-disable-next-line max-len
     const votingReputationClient: ExtensionClient = yield colonyManager.getClient(
       ClientType.VotingReputationClient,
       colonyAddress,
     );
-    const motion = yield votingReputationClient.getMotion(motionId);
 
-    const networkEstimate = yield provider.estimateGas({
-      from: votingReputationClient.address,
-      to:
-        /*
-         * If the motion target is 0x000... then we pass in the colony's address
-         */
-        motion.altTarget === AddressZero
-          ? colonyClient.address
-          : motion.altTarget,
-      data: motion.action,
+    const encodedFinalizeMotion = votingReputationClient.interface.functions[
+      'finalizeMotion(uint256)'
+    ].encode([motionId]);
+    const finalizeEstimate = yield provider.estimateGas({
+      from: userAddress,
+      to: votingReputationClient.address,
+      data: encodedFinalizeMotion,
     });
 
-    /*
-     * Increase the estimate by 100k WEI. This is a flat increase for all networks
-     *
-     * @NOTE This will need to be increased further for `setExpenditureState` since
-     * that requires even more gas, but since we don't use that one yet, there's
-     * no reason to account for it just yet
-     */
-    const estimate = bigNumberify(networkEstimate).add(bigNumberify(100000));
-
     const {
```

Comment:
The previous solution wasn't optimal, and I think it was done as a quick workaround. 
 
The problem is that it's not a reliable way of estimating the gas to finalize the motion. For example, with the safe transactions, I couldn't finalize them because the estimated gas limit was lower than what it required.

Alex suggested that instead of estimating the action inside the finalize function and then manually adding more gas we could estimate the execution of the "finalizeMotion" transaction itself and that would give a better, more accurate gas limit.

Reply:
Awesome, thanks